stardis/radiation_field/base.py

```diff
@@ -31,16 +32,50 @@ class RadiationField(HDFWriterMixin):
         calculate the total opacity at each frequency at each depth point.
     F_nu : numpy.ndarray
         Radiation field fluxes at each frequency at each depth point. Initialized as zeros and calculated by a solver.
+    thetas : numpy.ndarray
+        Theta angles for raytracing.
+    I_nus_weights : numpy.ndarray
+        Weights for the theta angles.
+    I_nus : numpy.ndarray
+        Radiation field intensity at each frequency at each depth point at each theta angle. Initialized as zeros and calculated by a solver.
+    track_individual_intensities : bool
+        Flag to track individual intensities at each theta angle. Default is False.
     """
 
     hdf_properties = ["frequencies", "opacities", "F_nu"]
 
-    def __init__(self, frequencies, source_function, stellar_model):
+    def __init__(
+        self,
+        frequencies,
+        source_function,
+        stellar_model,
+        num_of_thetas,
+        track_individual_intensities=False,
+    ):
         self.frequencies = frequencies
         self.source_function = source_function
         self.opacities = Opacities(frequencies, stellar_model)
         self.F_nu = np.zeros((stellar_model.no_of_depth_points, len(frequencies)))
 
+        # This uses gauss-legendre quadrature to sample thetas
+        thetas, weights = np.polynomial.legendre.leggauss(num_of_thetas)
+        self.thetas = (thetas / 2) + 0.5 * np.pi / 2
+        self.I_nus_weights = weights * np.pi / 2
+        self.track_individual_intensities = track_individual_intensities
+        if track_individual_intensities:
+            self.I_nus = np.zeros(
+                (stellar_model.no_of_depth_points, len(frequencies), len(self.thetas))
+            )
+
+        # This was our original theta sampling method
+        # dtheta = (np.pi / 2) / num_of_thetas  # Korg uses Gauss-Legendre quadrature here
```

Comment:
Does this need to be kept? If so I would put it in a test for comparison purposes.